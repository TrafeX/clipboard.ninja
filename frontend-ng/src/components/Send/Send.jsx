import {
  Box,
  Button,
  Flex,
  Heading,
  Text,
  Textarea,
  useColorModeValue,
  Icon,
} from "@chakra-ui/react";
import { useState } from "react";
import { IoBulbOutline } from "react-icons/io5";

const Send = ({ sendMessage, connectedToRoom, usersInRoom }) => {
  const [message, setMessage] = useState("");

  const subtextColor = useColorModeValue("gray.600", "gray.400");

  const sendThisMessage = () => {
    sendMessage(message);
    setMessage("");
  };

  const handleChange = (event) => {
    setMessage(event.target.value);
  };

  if (connectedToRoom === null) {
    return null;
  }

  return (
    <Box>
      <Box>
        <Heading as="h3" size="sm" mb={2}>
          Connected to Device
        </Heading>
        <Text fontSize={"sm"} color={subtextColor}>
          You are connected to the other {usersInRoom > 2 ? `${usersInRoom-1} devices` : 'device'} and can send text.
        </Text>
        <Text fontSize={"2xl"} fontWeight={"extrabold"} color={"brand.800"}>
          {connectedToRoom}
        </Text>
      </Box>
      <Flex py={2} direction="column">
        <Textarea
          placeholder="Enter the text to send"
          value={message}
          onChange={handleChange}
        />
        <Button
          type="submit"
          label="Send"
          colorScheme={"brand"}
          alignSelf="flex-start"
          my={2}
          onClick={() => sendThisMessage()}
          isDisabled={message.length === 0}
        >
          Send
        </Button>
      </Flex>
      <Text mt={2} fontSize={"sm"} color={subtextColor}>
        <Icon as={IoBulbOutline} />
        It&apos;s possible to send to multiple devices by also connecting them
        to the Device ID above.
      </Text>
    </Box>
  );
};

export default Send;
