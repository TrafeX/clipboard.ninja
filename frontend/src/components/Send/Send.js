import React, { useState } from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import muiThemeable from 'material-ui/styles/muiThemeable';

const style = {
  button: {
    margin: 12
  },
  textfield: {
    textAlign: 'left',
  }
};

type Props = {
  sendMessage: (string) => void,
  connectedToRoom: ?number,
  usersInRoom: number,
  muiTheme: Object,
};

const Send = ({ sendMessage, connectedToRoom, usersInRoom, muiTheme }: Props) => {

  const [message, setMessage] = useState('');

  const sendThisMessage = () => {
    sendMessage(message);
    setMessage('');
  };

  const handleChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
    setMessage(event.target.value);
  };


  if (connectedToRoom === null) {
    return null;
  }

  let moreUsersInRoom = null;
  if (usersInRoom > 2) {
    moreUsersInRoom = ` with ${usersInRoom - 1} devices`;
  }
  return (
    <div>
      <h3>
        Connected to Device ID:&nbsp;
        <span style={{ color: muiTheme.palette.accent1Color }}>
          <strong>
            {connectedToRoom}
          </strong>
        </span>
        {moreUsersInRoom}
      </h3>
      <p>
        You're connected to the other device(s) and can start sending text!<br />
        <small>It's possible to send to multiple devices by also connecting them to the Device Id above.</small>
      </p>
      <TextField
        floatingLabelText="Enter the text to send"
        multiLine={true}
        rows={1}
        value={message}
        style={style.textfield}
        onChange={handleChange}
      />
      <br />
      <RaisedButton
        label="Send"
        primary={true}
        style={style.button}
        onClick={() => sendThisMessage()}
        disabled={message === ''}
      />
    </div>
  );
}

export default muiThemeable()(Send);
