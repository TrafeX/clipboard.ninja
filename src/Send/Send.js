// @flow
import React, { Component } from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import muiThemeable from 'material-ui/styles/muiThemeable';

const style = {
  button: {
    margin: 12
  },
  textfield: {
    textAlign: 'left',
  }
};

class Send extends Component {
  props: {
    sendMessage: (string) => void,
    connectedToRoom: ?number,
    usersInRoom: number,
    muiTheme: Object,
  };
  state = {
    message: '',
  };

  sendMessage = () => {
    this.props.sendMessage(this.state.message);
    this.setState({
      message: '',
    })
  };

  handleChange = (event: Object) => {
    this.setState({
      message: event.target.value,
    })
  };

  render() {
    if (this.props.connectedToRoom === null) {
      return null;
    }
    let moreUsersInRoom = null;
    if (this.props.usersInRoom > 2) {
      moreUsersInRoom =  ` with ${this.props.usersInRoom-1} devices`;
    }
    return (
      <div>
        <h3>
          Connected to Device ID:&nbsp;
          <span style={{color: this.props.muiTheme.palette.accent1Color}}>
            <strong>
              {this.props.connectedToRoom}
            </strong>
          </span>
          {moreUsersInRoom}
        </h3>
        <p>
          You're connected to the other device(s) and can start sending text!<br />
          <small>It's possible to send to multiple devices by also connecting them to the Device Id above.</small>
        </p>
        <TextField
          floatingLabelText="Enter the text to send"
          multiLine={true}
          rows={1}
          value={this.state.message}
          style={style.textfield}
          onChange={this.handleChange}
        />
        <br />
        <RaisedButton
          label="Send"
          primary={true}
          style={style.button}
          onClick={() => this.sendMessage()}
          disabled={this.state.message === ''}
        />
      </div>
    );
  }
}

export default muiThemeable()(Send);
