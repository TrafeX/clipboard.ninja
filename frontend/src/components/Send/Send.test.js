import React from 'react';
import Send from './Send';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const wrapper = shallow(<Send/>).find('Send').shallow();
  expect(wrapper).toMatchSnapshot();
});
