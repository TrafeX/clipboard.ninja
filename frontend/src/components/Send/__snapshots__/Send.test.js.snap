// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h3>
    Connected to Device ID:Â 
    <span
      style={
        Object {
          "color": "#ff4081",
        }
      }
    >
      <strong />
    </span>
  </h3>
  <p>
    You're connected to the other device(s) and can start sending text!
    <br />
    <small>
      It's possible to send to multiple devices by also connecting them to the Device Id above.
    </small>
  </p>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Enter the text to send"
    fullWidth={false}
    multiLine={true}
    onChange={[Function]}
    rows={1}
    style={
      Object {
        "textAlign": "left",
      }
    }
    type="text"
    underlineShow={true}
    value=""
  />
  <br />
  <RaisedButton
    disabled={true}
    fullWidth={false}
    label="Send"
    labelPosition="after"
    onClick={[Function]}
    primary={true}
    secondary={false}
    style={
      Object {
        "margin": 12,
      }
    }
  />
</div>
`;
